src/Test.FunctionalTests.BinSkim.Rules/RuleTests.cs

```diff
@@ -277,13 +278,23 @@ private BinaryAnalyzerContext CreateContext(TestMessageLogger logger, Properties
                 context.Rule = skimmer;
 
                 AnalysisApplicability applicability = skimmer.CanAnalyze(context, out string reasonForNotAnalyzing);
+
                 if (applicability != expectedApplicability)
                 {
                     sb.AppendLine("CanAnalyze did not correct indicate target applicability (unexpected return was " +
                         applicability + "): " +
                         Path.GetFileName(target));
                     continue;
                 }
+
+                if (expectedReasonForNotAnalyzing != null && reasonForNotAnalyzing != expectedReasonForNotAnalyzing)
+                {
+                    sb.AppendLine("Cannot Analyze but unexpected reason identified (unexpected return was " +
```

Comment:
>Cannot [](http://example.com/codeflow?start=35&length=6)

So, this code follows the current convention of using the '+' to concatenate. You could convert to using interpolated strings instead. If you do, maybe clean up the rest of the file for interpolated string usage? e.g.:

Notice some other small clean-ups: 1) fixed casing of 'Analyze', 2) enquoted the dynamic injected values in apostrophes. This is very helpful in test output and I will offer this suggestion repeatedly. i.e., (We expected 'foo' but saw 'bar') is more readable than (We expected foo but saw bar)

                    sb.AppendLine(
                        $"Cannot analyze but unexpected reason identified (unexpected return was " +
                        $"'{reasonForNotAnalyzing}' but " +
                        $"'{expectedReasonForNotAnalyzing}' was expected): " +
                        Path.GetFileName(target));